Source/DafnyLanguageServer/Workspace/DafnyDocument.cs

```diff
@@ -71,6 +70,9 @@ public record DafnyDocument(
     }
 
     public int LinesCount => VerificationTree.Range.End.Line;
+    public IVerificationProgressReporter? GutterProgressReporter { get; set; }
+    public ConcurrentStack<Counterexample>? Counterexamples { get; set; }
+    public ConcurrentDictionary<ImplementationId, ImplementationView>? ImplementationViews { get; set; }
```

Comment:
Same here, we could have
```suggestion
    public ConcurrentDictionary<ImplementationId, ImplementationView>? ImplementationIdToView { get; set; }
```
to not only reflect what values are, but also what keys are.